src/Interpreters/Cache/LRUFileCachePriority.cpp

```diff
@@ -215,9 +215,20 @@ void LRUFileCachePriority::iterate(IterateFunc && func, const CachePriorityGuard
                 ErrorCodes::LOGICAL_ERROR,
                 "Mismatch of file segment size in file segment metadata "
                 "and priority queue: {} != {} ({})",
-                entry.getSize(), metadata->size(), metadata->file_segment->getInfoForLog());
+                entry.getFilledSize(), metadata->size(), metadata->file_segment->getInfoForLog());
         }
 
+        if (metadata->getEntrySize() != entry.getSize())
```

Comment:
This got overcomplicated, let's instead add to each `metadata->getSize()` and `entry.getSize()` an argument `bool aligned` (`getSize(true/false)`) instead of new methods `getEntrySize()` and`getFilledSize()`